tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -1326,3 +1327,127 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST_RampUp_NoLeap)
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60,                      !- Minutes per item",
+        "  Yes;                     !- Adjust Schedule for Daylight Savings",
+        " ",
+        "Schedule:File,",
+        "  Test2,                   !- Name",
+        "  ,                        !- Schedule Type Limits Name",
+        "  " + scheduleFile.string() + ",              !- File Name",
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60,                      !- Minutes per item",
+        "  No;                     !- Adjust Schedule for Daylight Savings",
+        " ",
+        "Schedule:File,",
+        "  Test3,                   !- Name",
+        "  ,                        !- Schedule Type Limits Name",
+        "  " + scheduleFile.string() + ",              !- File Name",
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60,                      !- Minutes per item",
+        "  ;                     !- Adjust Schedule for Daylight Savings",
+        " ",
+        "Schedule:File,",
+        "  Test4,                   !- Name",
+        "  ,                        !- Schedule Type Limits Name",
+        "  " + scheduleFile.string() + ",              !- File Name",
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60;                      !- Minutes per item",
+        " ",
+    });
+
+    // This will process the provided idf chunk within the test fixture (must pass this step in order to proceed)
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->MinutesPerTimeStep = 60;
+    state->dataGlobal->TimeStep = 1;         // Checking to see if omitting this is OK here
+    state->dataEnvrn->DayOfWeek = 1;         // Sunday
+    state->dataEnvrn->DayOfWeekTomorrow = 2; // Monday
+    state->dataEnvrn->DayOfYear_Schedule = 1;
+    state->dataGlobal->HourOfDay = 24;
+
+    // Test 1 condition
```

Comment:
Testing four different conditions each starting with a comment line that says "// Test X condition "...